libraries/conf_helpers.rb

```diff
@@ -27,35 +39,74 @@ def self.read_config(conf_path)
     end
 
     def self.parse_config(conf_body)
-      current_stanza = 'default'
+      current_section = 'default'
       current_config = {}
 
       conf_body.each_line do |ln|
         case ln
         when /^\s*\[([^#]+)\]\s*/
-          current_stanza = Regexp.last_match[1]
+          current_section = Regexp.last_match[1]
         when /^\s*([^#]+?)\s*=\s*([^#]+?)\s*(?:#.*)?$/
-          (current_config[current_stanza] ||= {})[Regexp.last_match[1]] = Regexp.last_match[2]
+          (current_config[current_section] ||= {})[Regexp.last_match[1]] = Regexp.last_match[2]
         end
       end
 
       current_config
     end
 
     def self.merge_config(current_config, desired_config)
```

Comment:
10f9c8ad238395f5f34f8d298e122ec93ce26cda

I split the nil filtering into a new method, filter_config, and call it from the conf resource. Also the file writing logic has been moved to a template in the conf resource and merge_config now returns hash.